### PR TITLE
Implement the createQuery overriding

### PR DESCRIPTION
**Pull request by chervaliery**
The proxyQuery passed to the Datagrid was hard coded. Here, it check if the createQuery is overridden by the user and return an ElasticaProxyQuery and if it’s a smart query. Else, the original model manager is used instead of ElasticSearch.

### CODE REVIEW

File: Builder/ElasticaDatagridBuilder.php [hunk 1]
```diff
@@ -106,9 +106,15 @@ public function getBaseDatagrid(AdminInterface $admin, array $values = array())
...
 
-        $proxyQuery = new ElasticaProxyQuery(
-            $this->finderProvider->getFinderByAdmin($admin)
-        );
+        $proxyQuery = $admin->createQuery ();
+        // if the default modelmanager query builder is used, we need to replace it with elastica
+        // if not, that means $admin->createQuery has been overriden by the user and return already an ElasticaProxyQuery object
+        if (! $query instanceof Sonata\AdminSearchBundle\ProxyQuery\ElasticaProxyQuery) {
+            $issmart = $this->isSmart ( $admin, $values );
+            if ($issmart) {
```
Comment (greg0ire) [inline Builder/ElasticaDatagridBuilder.php]: Please do not create variables that are used only once. If you need to clarify what something does, please use comments. In that case, the name of the variable and the method is the same so you can simply do : `if ($this->isSmart($admin, $values))`.

File: Builder/ElasticaDatagridBuilder.php [hunk 2]
```diff
@@ -106,9 +106,15 @@ public function getBaseDatagrid(AdminInterface $admin, array $values = array())
             $defaultOptions
         );
 
-        $proxyQuery = new ElasticaProxyQuery(
-            $this->finderProvider->getFinderByAdmin($admin)
-        );
+        $proxyQuery = $admin->createQuery ();
```
Comment (greg0ire) [inline Builder/ElasticaDatagridBuilder.php]: I believe this violates PSR. Please make sure your IDE is PSR-aware. There are many other violations in this PR I think.

File: Builder/ElasticaDatagridBuilder.php [hunk 3]
```diff
@@ -148,6 +154,17 @@ public function isSmart(AdminInterface $admin, array $values = array())
             }
 
             if (!in_array($key, $mappedFieldNames)) {
+                /*
+                 * We are in the case where a field is used as filter without being mapped in elastic search.
+                 * An ugly case would be to have a custom field used in the filter without mapping in the model, we need to control that
+                 */
+                $ret = $admin->getModelManager ()->getParentMetadataForProperty ( $admin->getClass(), $key, $admin->getModelManager());
```
Comment (greg0ire) [inline Builder/ElasticaDatagridBuilder.php]: I think this line exceeds 120 chars.

**Comment by chervaliery** [comment]
I fixed the code to respect PSR, but I fail to merge with the master. I will re create a pull request

**Comment by greg0ire** [comment]
Do you know that you can update pull request by force pushing to the branch that corresponds ?

**Pull request closed without merging**